packages/zoe/src/contracts/gimix/gimix.js

```diff
@@ -0,0 +1,113 @@
+// @ts-nocheck
+/* eslint-disable no-unused-vars */
+import { prepareExoClass } from '@agoric/vat-data';
+import { M } from '@endo/patterns';
+import { OfferHandlerI } from '../../typeGuards';
+
+/**
+ * @typedef {object} GiMiXTerms
+ * @property {import('@agoric/vats').NameHub} namesByAddress
+ */
+
+const JobReportProposalShape = M.splitRecord({
+  give: {},
+  want: {},
+  exit: { onDemand: undefined },
+});
+
+const OracleInvitationProposalShape = M.splitRecord({
+  give: {},
+  want: {},
+  exit: { onDemand: undefined },
+});
+
+const WorkAgreementProposalShape = M.splitRecord({
+  give: {},
+  want: {},
+  exit: { onDemand: undefined },
+});
+
+/**
+ * @param {ZCF<GiMiXTerms>} zcf
+ */
+export const prepare = async (zcf, _privateArgs, baggage) => {
+  const { namesByAddress } = zcf.getTerms();
+
+  const makeJobReportInvitationHandler = prepareExoClass(
+    baggage,
+    'JobReportHandler',
+    OfferHandlerI,
+    () => ({}),
+    {
+      handle(seat) {
+        const { self, state } = this;
+        return 'something';
+      },
+    },
+  );
+
+  const makeOracleInvitationHandler = prepareExoClass(
+    baggage,
+    'OracleInvitationHandler',
+    OfferHandlerI,
+    () => ({}),
+    {
+      handle(seat) {
+        const { self, state } = this;
+        const jobReportInvitationMaker = makeJobReportInvitationHandler();
+        return jobReportInvitationMaker;
+      },
+    },
+  );
+
+  const makeWorkAgreementInvitationHandler = prepareExoClass(
+    baggage,
+    'WorkAgreementHandler',
+    OfferHandlerI,
+    () => ({}),
+    {
+      handle(seat) {
+        const { self, state } = this;
+        const jobId = 'placeholder42';
+        return jobId;
+      },
+    },
+  );
+
+  const makeJobReportInvitation = () => {
+    const jobReportInvitationHandler = makeJobReportInvitationHandler();
+
+    const jobReportInvitationP = zcf.makeInvitation(
+      jobReportInvitationHandler,
+      'gimix job report',
+      undefined,
+      JobReportProposalShape,
+    );
+    return jobReportInvitationP;
+  };
+
+  const makeOracleInvitation = () => {
+    const oracleInvitationHandler = makeOracleInvitationHandler();
+
+    const oracleInvitationP = zcf.makeInvitation(
+      oracleInvitationHandler,
+      'gimix oracle invitation',
+      undefined,
+      OracleInvitationProposalShape,
+    );
+    return oracleInvitationP;
```

Comment:
This doesn't quite fit the continuing invitation pattern supported by the smart wallet.

_narrator: no wonder; documentation is lacking, to say the least._
  - https://github.com/Agoric/documentation/issues/667

The offer result has to be an object with an `invitationMakers` property with methods for making further invitations; for example:

```js
      invitationMakers: {
        AdjustBalances() {
          return this.facets.holder.makeAdjustBalancesInvitation();
        },
        CloseVault() {
          return this.facets.holder.makeCloseInvitation();
        },
        TransferVault() {
          return this.facets.holder.makeTransferInvitation();
        },
      },
```
https://github.com/Agoric/agoric-sdk/blob/5dc325b8cc1d94f46ecc18a7c67a14989feade11/packages/inter-protocol/src/vaultFactory/vaultHolder.js#L73C1-L83C9

Reply:
At https://github.com/Agoric/agoric-sdk/blob/5dc325b8cc1d94f46ecc18a7c67a14989feade11/packages/governance/src/voterKit.js#L44-L45

Reply:
I _think_ the method names showed up in one of our UIs at some point. @dtribble do you remember?

Reply:
@dckc , PTAL. Am I following the pattern you have in mind? Are there more conventions I should know about?

Reply:
IIUC, this line is not where your suggestion applies. Rather, it applies to lines 57-58 above, which I just corrected to try to follow the continuing invitation pattern. Yes?

Reply:
Look at the second commit for the differential from the one you reviewed.

Reply:
yes, that looks right, I think.

Reply:
Thx. Resolving.